### DISCUSSION TITLE
Info regarding image size

### DISCUSSION THREAD

**Discussion by Sarath-C-Koppolu** [ultralytics/ultralytics#13438] [General]
Hello thank you so much for earlier response. 

I have 2 questions,

As i already asked you this that training image with 1280 and val with different image sizes. 

Is it feasible to use in such a manner or training with different images is better. 

I mean will there be any huge difference in mAP and inference time..?

And when can I expect yolov10 from ultralytics?

Thank you

**Comment by glenn-jocher** [comment]
@Sarath-C-Koppolu hello! 😊

You're very welcome! Let's dive into your questions:

### 1. Training and Validation with Different Image Sizes

Training your model with images of size 1280 and validating with different image sizes is indeed feasible. However, there are a few considerations to keep in mind:

- **Consistency**: Keeping the image size consistent between training and validation can lead to more predictable performance metrics. If you train with 1280 and validate with a smaller size, the model might not perform optimally since it hasn't seen smaller images during training.
- **mAP (Mean Average Precision)**: There might be a slight drop in mAP if the validation image size is significantly different from the training size. This is because the model's learned features are optimized for the training image size.
- **Inference Time**: Smaller image sizes generally lead to faster inference times. If you validate with smaller images, you might notice quicker inference but potentially at the cost of some accuracy.

Here's a quick example of how you can set different image sizes for training and validation using the Ultralytics YOLOv8:

```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.pt')

# Train the model with image size 1280
model.train(data='coco128.yaml', epochs=100, imgsz=1280)

# Validate the model with a different image size, e.g., 640
metrics = model.val(imgsz=640)
```

### 2. YOLOv10 Release

As for YOLOv10, it's an exciting topic! 🚀 While I don't have a specific release date to share, the Ultralytics team is always working on pushing the boundaries of what's possible in real-time object detection. Keep an eye on our [official announcements](https://ultralytics.com) and GitHub releases for the latest updates!

Thank you for your questions, and happy experimenting with YOLO! If you have any more queries or need further assistance, feel free to ask.
